### DISCUSSION TITLE
More Animation Script API

### DISCUSSION THREAD

**Discussion by TartaricAcid** [TartaricAcid/TouhouLittleMaid#325] [Suggestions & Ideas]
1. Unique Random Seed
  
    Get a fixed value, each maid is different, similar to the entity's UUID
    
    ```js
    long maid.getSeed()  // get maid's uuid least significant bits
    ```
    
2. Jump Check

    Check if the current entity is in a jumping state

    ```js
    boolean maid.isJump()
    ```

3. Height Check

    Check the distance between the current entity and the block below it

    ```js
    double maid.getDistanceBelow()
    ```

**Comment by SAlexW** [comment]
`int maid.getDistanceBelow()`
I think, it can be "float" or "double", because maid can fall from half block, so the height at the beginning of the fall can be 0.4-0.5 (bed or half-blocks), 0.6-0.8 (enchanting table, chest or any non-full size blocks), 1.5 (fence), 2.5 (fence on block) and so on. Also, any entity falls smoothly, so the drop height changes at each moment by a small amount, so the height of 2.435 is quite possible.

**Reply by TartaricAcid** [reply]
Thanks for correcting
